### ISSUE TITLE
Resource availability on PowerShell gallery

### ISSUE THREAD

**Issue by anwather** [PowerShell/JEA#33]
Can we get this put on the PowerShell gallery? @rpsqrd

**Comment by rpsqrd** [comment]
I don't have any capacity right now to drive that work, but I'd be in full support of onboarding this to the [DSC Resource Kit](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md), which would help it get published on the gallery (and increase visibility). Is that something you could help with?

**Comment by anwather** [comment]
Would like to - just have to find some time - maybe Q4

**Comment by hackjammer** [comment]
As a potential user - it makes it much more difficult to use if it's not easily downloadable format - getting this properly supported and in the gallery is important, particularly given that it is a recommended way of setting up JEA in the official documentation.

**Comment by jnury** [comment]
@rpsqrd There is a xJEA DSCResource in the resource kit: https://github.com/PowerShell/xJea

It seems to me that it's an 'old' fashion to build JEA Endpoint (aka pre-PS5). How can we deal with it ?

I'm back with some spare time. I'll start with fixing some openend issues, and after that I'll try to bring up a decent release pipeline (I have to figure out how to do integration tests with an AD and AppVeyor).

**Comment by rpsqrd** [comment]
You're correct -- it's the proof of concept that led to the JEA we know today. We're okay to pave the xJea resource and replace it with a new, PS 5.1+ implementation. We just need to be clear in the PR that it's a major version and breaking change. We can update the README with information about how to download the old version with the -RequiredVersion parameter if someone wants the previous implementation.

Alternatively, we could just go for a proper DSC resource (e.g. JeaDsc) that meets all of the [high quality DSC resource requirements](https://github.com/PowerShell/DscResources/blob/master/HighQualityModuleGuidelines.md). I'm fine either way.

**Comment by anwather** [comment]
HQRM looks to be the way to go

**Comment by anwather** [comment]
I have a bit of spare time for this - I can at least set up the template resource with some basic tests - I might just create a new repo for this and then can merge it all later -

**Comment by rpsqrd** [comment]
I'm following up with the DSC team to see if they have any preference for how we release a new resource. I'll keep this thread posted and get some of the logistics set up once I have an answer.

One change @commonsensesoftware suggested that I think is great is to add the "Ensure = Present|Absent" property so we can help people _un_configure JEA if they need to.

I'm also interested in hearing everyone's thoughts on also adding a role capability resource so users don't have to create them out of band and copy the files to the target machines.

**Comment by jnury** [comment]
Ensure = Present|Absent -> Yes !
JEARoleCapabilitiesDSC -> Yes !

**Comment by jnury** [comment]
@rpsqrd : logistics = build process and Unit Testing VMs ? :-)
